wolfcrypt/src/port/kcapi/kcapi_ecc.c

```diff
@@ -190,14 +192,52 @@ int KcapiEcc_Sign(ecc_key* key, const byte* hash, word32 hashLen, byte* sig,
         }
     }
     if (ret == 0) {
-        ret = kcapi_akcipher_sign(key->handle, hash, hashLen, sig, *sigLen,
+        if ((long int)sig % sysconf(_SC_PAGESIZE) != 0) {
+            ret = posix_memalign((void*)&sig_aligned,
+                    sysconf(_SC_PAGESIZE), *sigLen);
+            if (ret < 0) {
+                ret = MEMORY_E;
+            }
+        }
+        else {
+            sig_aligned = sig;
```

Comment:
Seems like it would make sense to have `hash_aligned` not be `const`.... and cast?

Reply:
I didn't want to cast away the const-ness of the original hash pointer. The pointer being to const data is supposed to offer the compiler some chance of optimization when the data is supposed to be unchanging. I guess since we know what is using this data isn't going to change it, we can reduce the number of casts.

Reply:
I made this change after all.